metaseq/checkpoint_utils.py

```diff
@@ -97,12 +97,56 @@ def save_checkpoint(
             async_callback_fn=async_callback_fn if save_to_NFS else None,
         )
 
+        for cp in checkpoints[1:]:
+            assert PathManager.copy(checkpoints[0], cp, overwrite=True), f"Failed to copy {checkpoints[0]} to {cp}"
+
         write_timer.stop()
         logger.info(
             f"Saved checkpoint {checkpoints[0]} (epoch {epoch} @ {updates} updates) "
             f"(writing took {write_timer.sum} seconds)"
         )
 
+    delete_old_checkpoint_files(cfg, end_of_epoch, suffix, trainer.is_data_parallel_master)
```

Comment:
Ah I took out default checkpoint deletion logic since we are checkpointing quite often for larger runs, but having to restart from much older checkpoints when things start looking unhealthy.

Let me work on splitting this PR up!

Reply:
Plopped into https://github.com/facebookresearch/metaseq/pull/679